### DISCUSSION TITLE
Dumping/loading schema in binary format

### DISCUSSION THREAD

**Discussion by pkmuthu83** [CESNET/libyang#2322] [Q&A]
Is it possible to dump the libyang schema/context (which is already loaded in memory) in LYB format, similar to how data is dumped ? Cannot find API for it, hence asking. Also, we want to subsequently load the schema in binary format rather than text format. Thanks !

**Comment by michalvasko** [comment]
There is no such format for YANG modules, only YANG and YIN.

**Comment by pkmuthu83** [comment]
Thanks. I am looking for a way to load yang schema files really fast. If we have large number of yang files and we parsed the yang files once into libyang context, is there a way to dump the in-memory context in a format which can be loaded very fast the next time..

**Reply by michalvasko** [reply]
No, there is no such possibility. But we are currently working on a feature that should allow this although it is still months away from being ready.

**Reply by pkmuthu83** [reply]
HI @michalvasko  : Any update on this feature ?

**Reply by michalvasko** [reply]
Actually, there is. The feature will be different but should server your purpose, even better than your idea. We are working on support for in-memory context that can be directly used by libyang. Meaning you should be able to "print" the created context once and then just use it until you need to make some changes to it. No parsing involved anymore.

**Reply by pkmuthu83** [reply]
Ok, thanks. Any idea on the timelines for this feature ?

**Reply by michalvasko** [reply]
Not really, the feature is rather complex and while almost finished, it will need a lot of testing. So a few months maybe.

**Reply by pkmuthu83** [reply]
@michalvasko Just wanted to check back on this feature (i.e dumping/printing the context once it is parsed and re-use later instead of parsing again).

**Reply by michalvasko** [reply]
Right, sorry for not replying here, I have completely forgotten there is such a discussion. The feature has been implemented for some time and seems to be working well but it is not that simple to use. Look into the [docs](https://netopeer.liberouter.org/doc/libyang/master/html/howto_context.html) for more information about the Printed Context.

**Reply by pkmuthu83** [reply]
Thank you
